providers/fips/self_test_data.inc

```diff
@@ -1651,1913 +1646,1832 @@ static const ST_KAT_PARAM dsa_key[] = {
-    0x1f, 0x45, 0xe1, 0xf6, 0x65, 0x1f, 0xba, 0xe4,
-    0xf6, 0xca, 0xa9, 0x62, 0xa9, 0x2c, 0x3a, 0x2d,
-    0x8d, 0x34, 0xe2, 0x32, 0x6e, 0x4a, 0x52, 0x9f,
-    0x3d, 0xcf, 0xab, 0xe8, 0x63, 0x66, 0x3f, 0x06,
-    0xa3, 0xaa, 0xd5, 0xcb, 0x5d, 0x41, 0xb8, 0xe7,
-    0x0a, 0x6b, 0x45, 0x90, 0x3a, 0xf0, 0xe5, 0x7f,
-    0x7e, 0xde, 0x68, 0x20, 0x5e, 0x5a, 0x31, 0x3a,
-    0x5c, 0x25, 0xb3, 0x82, 0xc7, 0x1e, 0x7d, 0x4d,
-    0xd0, 0x23, 0x9f, 0x22, 0x1a, 0x54, 0x0b, 0xe4,
-    0x3c, 0x1c, 0xc5, 0x24, 0x39, 0x4d, 0x96, 0x90,
-    0xce, 0xc9, 0x3a, 0x07, 0x45, 0x01, 0x9e, 0xb6,
-    0x55, 0x1b, 0xdc, 0xe0, 0x3b, 0xe8, 0x92, 0x57,
-    0x43, 0x12, 0x9b, 0x11, 0x3f, 0x81, 0x5a, 0xca,
-    0xc5, 0x85, 0x39, 0x25, 0x6d, 0xe4, 0x30, 0xef,
-    0x83, 0x21, 0xed, 0x9c, 0xb3, 0xcf, 0x10, 0xe6,
-    0xa1, 0x46, 0x10, 0x9a, 0x7b, 0xaf, 0x5f, 0x75,
-    0x99, 0x62, 0xc1, 0xf4, 0x2c, 0x28, 0x0a, 0x8a,
-    0xc7, 0xd2, 0xb4, 0x75, 0xb9, 0x66, 0x70, 0x9a,
-    0xaf, 0xe3, 0xf1, 0x2f, 0xf2, 0xb0, 0x59, 0x9c,
-    0x1e, 0x1a, 0xbd, 0xa9, 0x15, 0x55, 0x95, 0x4c,
-    0x5b, 0x78, 0x0b, 0x2c, 0x00, 0xbb, 0xf9, 0x8b,
-    0xeb, 0x72, 0x4f, 0xfb, 0xa4, 0x3a, 0xa0, 0x04,
-    0x45, 0x32, 0x02, 0x6c, 0x16, 0x52, 0x3f, 0x4a,
-    0x0a, 0x77, 0x64, 0xf4, 0x74, 0xed, 0x60, 0x6b,
-    0x5e, 0x43, 0xa7, 0xe3, 0x84, 0x7e, 0xda, 0xf0,
-    0xb1, 0x9e, 0x6d, 0x90, 0x9c, 0x32, 0xae, 0xba,
-    0x7c, 0xfb, 0x72, 0x22, 0x27, 0x9c, 0xe1, 0x85,
-    0xe1, 0xf6, 0x22, 0x4c, 0x3c, 0x4a, 0xd6, 0xed,
-    0x4c, 0xa7, 0x79, 0x59, 0xb5, 0x5b, 0x91, 0x65,
-    0x3f, 0x93, 0x97, 0x8d, 0xd7, 0xd0, 0xab, 0x17,
-    0x2f, 0x13, 0x74, 0x53, 0x69, 0x74, 0xf8, 0x6b,
-    0x39, 0x5c, 0x64, 0x5b, 0x3d, 0x75, 0xca, 0x85,
-    0x0e, 0xda, 0x0f, 0x01, 0x34, 0xa3, 0x67, 0x8e,
-    0x26, 0x6f, 0x26, 0x3c, 0xd0, 0xd9, 0xae, 0xe8,
-    0x52, 0x13, 0x7f, 0xf8, 0x69, 0x62, 0xfc, 0x9a,
-    0xc0, 0x0c, 0x66, 0x2e, 0x57, 0x21, 0x75, 0xb0,
-    0xb3, 0x8c, 0xf6, 0x97, 0x44, 0x46, 0x65, 0x15,
-    0x79, 0xd5, 0x6b, 0x68, 0x96, 0x47, 0xc1, 0xba,
-    0x75, 0x46, 0x87, 0x76, 0x7d, 0x2d, 0xac, 0xf3,
-    0x16, 0xae, 0xfb, 0x7e, 0x41, 0xe4, 0xae, 0x15,
-    0xc2, 0x51, 0x69, 0x71, 0x0b, 0x63, 0x20, 0x6a,
-    0xbd, 0xad, 0xce, 0x2a, 0x94, 0xac, 0xcf, 0x15,
-    0x4e, 0xdc, 0x8e, 0x2a, 0x48, 0xed, 0xb3, 0x48,
-    0x95, 0xf4, 0x41, 0xf3, 0x52, 0xef, 0x62, 0x90,
-    0x10, 0x30, 0x42, 0xec, 0xf9, 0x30, 0x25, 0xc3,
-    0xc5, 0x47, 0x76, 0xb5, 0x37, 0xeb, 0x9e, 0x87,
-    0xbe, 0x5c, 0x24, 0xa5, 0x34, 0xdd, 0x92, 0xa1,
-    0x20, 0x7d, 0xa7, 0x94, 0xa2, 0x67, 0x26, 0x70,
-    0xfe, 0xc9, 0x3f, 0x21, 0xd1, 0xc2, 0x5a, 0xb1,
-    0xaa, 0xac, 0x14, 0x1a, 0xe4, 0xdb, 0x71, 0x7e,
-    0x9c, 0xc2, 0x52, 0x59, 0xc9, 0x58, 0xca, 0x88,
-    0x71, 0x4f, 0x90, 0xb1, 0xbb, 0xac, 0x80, 0x53,
-    0x21, 0xfb, 0xff, 0xfd, 0x1e, 0x9f, 0xc0, 0x59,
-    0x8d, 0x80, 0x8e, 0x85, 0xa9, 0x3b, 0xd3, 0x20,
-    0x3b, 0x91, 0x16, 0x49, 0xb2, 0x2a, 0xa4, 0xe2,
-    0x30, 0xb2, 0x76, 0xf6, 0x92, 0x71, 0x37, 0x1b
-};
+#ifndef OPENSSL_NO_ML_KEM
```

Comment:
The diff seem to be getting confused.  The change was to move the data upwards in the file, convert to ST_KAT_PARAM arrays and add wrappers for the test harness.

Reply:
@vdukhovni sorted the diff out.